cmd/influxd/upgrade/upgrade.go

```diff
@@ -44,72 +79,293 @@ var options = struct {
+		return err
+	}
+	if err := checkParam("password", options.target.password); err != nil {
+		return err
+	}
+	if err := checkParam("org", options.target.orgName); err != nil {
+		return err
+	}
+	if err := checkParam("bucket", options.target.bucket); err != nil {
+		return err
+	}
+
+	if options.source.configFile != "" {
+		log.Info("upgrading config file", zap.String("file", options.source.configFile))
+		if _, err := os.Stat(options.source.configFile); err != nil {
+			return err
+		}
+		v1Config, err := upgradeConfig(options.source.configFile, options.target, log)
+		if err != nil {
+			return err
+		}
+		options.source.metaDir = v1Config.Meta.Dir
+		options.source.dataDir = v1Config.Data.Dir
+		options.source.walDir = v1Config.Data.WALDir
+	} else {
+		log.Info("no InfluxDB 1.x config file specified, skipping its upgrade")
+	}
+
+	if err := options.source.checkDirs(); err != nil {
+		return err
+	}
+
+	log.Info("source paths", zap.String("meta", options.source.metaDir), zap.String("data", options.source.dataDir))
+	log.Info("target paths", zap.String("bolt", options.target.boltPath), zap.String("engine", options.target.enginePath))
+
+	if fi, err := os.Stat(options.target.enginePath); err == nil {
+		if !fi.IsDir() {
+			return fmt.Errorf("engine path '%s' is not directory", options.target.enginePath)
+		}
+		entries, err := ioutil.ReadDir(options.target.enginePath)
+		if err != nil {
+			return err
+		}
+		if len(entries) > 0 {
+			return fmt.Errorf("target engine path '%s' must be empty", options.target.enginePath)
+		}
+	}
 
 	v1, err := newInfluxDBv1(&options.source)
 	if err != nil {
 		return err
 	}
-	_ = v1
 
-	v2, err := newInfluxDBv2(ctx, &options.target)
+	v2, err := newInfluxDBv2(ctx, &options.target, log)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := v2.close(); err != nil {
+			log.Error("2.x services closing error", zap.Error(err))
+		}
+	}()
+
+	or, err := setupAdmin(ctx, v2)
+	if err != nil {
+		return err
+	}
+	options.target.token = or.Auth.Token
+
+	db2BucketIds, err := upgradeDatabases(ctx, v1, v2, or.Org.ID, log)
 	if err != nil {
+		//remove all files
+		log.Info("database upgrade error, removing data")
+		if e := os.Remove(options.target.boltPath); e != nil {
+			log.Error("cleaning failed", zap.Error(e))
+		}
+
+		if e := os.RemoveAll(options.target.enginePath); e != nil {
+			log.Error("cleaning failed", zap.Error(e))
+		}
 		return err
 	}
-	_ = v2
 
-	// 1. Onboard the initial admin user
-	// v2.onboardSvc.OnboardInitialUser()
+	if err = generateSecurityScript(v1, db2BucketIds, log); err != nil {
+		return err
+	}
 
-	// 2. read each database / retention policy from v1.meta and create bucket db-name/rp-name
-	// newBucket := v2.ts.CreateBucket(ctx, Bucket{})
-	//
-	// 3. create database in v2.meta
-	// v2.meta.CreateDatabase(newBucket.ID.String())
-	// copy shard info from v1.meta
+	log.Info("upgrade successfully completed. Start service now")
```

Comment:
nit: Log messages should be complete sentences and start with a capital. Please see the [logging style guide](https://github.com/influxdata/influxdb/blob/master/logger/style_guide.md)